03-rnaseq/dimension_reduction_rnaseq_01_pca.Rmd

```diff
@@ -0,0 +1,368 @@
+
+# Create a density plot using the sums
+ggplot2::ggplot(count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+    color = "blue",
+    linetype = "solid"
+  ) +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count)), # This line represents the mean count value
+    color = "red",
+    linetype = "dashed"
+  )
+```
+
+It appears that our arbitrary cutoff of 10 (the solid blue line) is a decent cutoff to make here as it seems to be just about inline with the mean counts value (the red dashed line) and retains the data points that are not in the majority and therefore may hold significance.
+
+```{r}
+# Define a minimum counts cutoff and filter `DESeqDataSet` object to include
+# only rows that have counts above the cutoff
+genes_to_keep <- rowSums(counts(dds)) >= 10
+dds <- dds[genes_to_keep, ]
+```
+
+## Perform DESeq2 normalization
+
+We are going to use the `vst()` function from the `DESeq2` package to normalize the data.
+TODO: Add a note/link to section in documentation on why we use `vst()` vs `rlog()` and `normTransform()`.
+
+```{r}
+# Normalize the data in the `DESeqDataSet` object using the `vst()` function from the `DESEq2` R package
+dds_norm <- vst(dds)
+```
+
+
+## Create PCA plot using DESeq2
+
+In this code chunk, the variable `refinebio_treatment` is given to the `plotPCA()` function as part of the goal of the experiment is to analyze the differential responses to androgen deprivation therapy (ADT).
+
+```{r}
+plotPCA(dds_norm,
+  intgroup = "refinebio_treatment"
+)
+```
+
+In this chunk, we are going to add a variable to our plot for annotation.
+
+The variables `refinebio_treatment` and `refinebio_disease` are given to the `plotPCA()`
+due to the hypothesis of the experiment stating "that differences in tumor transcriptional programs may reflect differential responses to androgen deprivation therapy (ADT) and subsequent metastasis".
+
+```{r}
+plotPCA(dds_norm,
+  intgroup = c("refinebio_treatment", "refinebio_disease")
+)
+```
+
+It's hard to distinguish the different `refinebio_treatment` versus `refinebio_disease` in the plot above.
+
+Let's use the `ggplot2` package functionality to customize our plot further.
+
+```{r}
+# We first have to save the results of the `plotPCA` function for use with `ggplot2`
+pcaData <-
+  plotPCA(
+    dds_norm,
+    intgroup = c("refinebio_treatment", "refinebio_disease"),
+    returnData = TRUE # This argument tells R to return the PCA values
+  )
+
+# Now let's plot using `ggplot2` functionality
+ggplot(pcaData, aes(
+  x = PC1,
+  y = PC2,
+  color = refinebio_treatment,
+  shape = refinebio_disease
+)) +
+  geom_point(size = 3)
+```
+
+## Save annotated PCA plot as a PNG
+
+You can easily switch this to save to a JPEG or TIFF by changing the function and file name within the function to the respective file suffix.
+
+```{r}
+# Open a PNG file
+png(file.path(
```

Comment:
We should use `ggsave()` here. (It can do PNGs if that was the concern?)

Reply:
Also may want to specifically name the plot (aka don't depend on default of `last_plot()`) since there's multiple plots and people may get confused which one is being save).